optax/_src/transform.py

```diff
@@ -953,6 +953,57 @@ def update_fn(updates, state, params=None):
   return base.GradientTransformation(init_fn, update_fn)
 
 
+class ScaleByOnlineNewtonStepState(NamedTuple):
+  """State holding the inverse of the sum of gradient outer products to date."""
+  hessian_inv: base.Updates
+
+
+def sherman_morrison(a_inv, u):
```

Comment:
Could you add type annotations to this function please? I assume, here this is just `chex.Array` for all inputs and the return type?

(Not all of optax was typed in the past but we'd like all new contributions to use type annotations and are slowly adding them to the old code too.)